packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1066,10 +1092,26 @@ function useMutableSource<Source, Snapshot>(
   // In both cases, we need to throw away pending udpates (since they are no longer relevant)
   // and treat reading from the source as we do in the mount case.
   if (
+    !is(prevGetSnapshot, getSnapshot) ||
     !is(prevSource, source) ||
-    !is(prevSubscribe, subscribe) ||
-    !is(prevGetSnapshot, getSnapshot)
+    !is(prevSubscribe, subscribe)
```

Comment:
This is neat that this is now the same condition that the effect hook uses when comparing deps. In the future we could  "cheat" and read the deps off the effect hook instead of using a ref.